### PR TITLE
Screen Kit mechanism to ensure display orientation consistency

### PR DESCRIPTION
**Pull request by radujipa**
Reviewable preliminary version of the PR, currently testing the approach.

## Ensure display_rotate is set globally for Screen Kits
When users (un)tick the Display > Flip option when using SKs, the
display_rotate config option is mirrored globally. This is to prevent
situations when SKs lose EDID that the screen orientation changes.

## Ensure screen kit orientation is set globally on first boot
The display_rotate option is already set for all SKs in the config
file using EDID filters. This ensures that the global setting also
matches the screen rotation for SKs to ensure consistency in the
unlikely event an SK loses EDID.

### CODE REVIEW

File: bin/kano-settings-onboot [hunk 1]
```diff
@@ -217,6 +219,22 @@ def ensure_correct_volume_levels(dry_run=False):
...
 
+def ensure_screen_kit_rotation_global():
+    """
+    Ensure the screen kit orientation setting is also set globally.
+
+    This sets a 'fallback flag' in the unlikely situation where a SK will
+    lose its EDID data. No reboot is required since the option is already set
+    in the config.txt by default - this just mirrors it.
+    """
+    if not get_config_value('display_rotate') and is_screen_kit():
```
Comment (Ealdwulf) [inline bin/kano-settings-onboot]: `is_screen_kit` uses EDID. What if that's already corrupted?

Comment (radujipa) [reply bin/kano-settings-onboot]: If the EDID is not there even at first boot, I'm throwing in the towel. It shouldn't pass manufacturing QA!

Comment (Ealdwulf) [reply bin/kano-settings-onboot]: As per verbal discussion - there is - we could check the USB.

File: bin/kano-settings-onboot [hunk 2]
```diff
@@ -217,6 +222,26 @@ def ensure_correct_volume_levels(dry_run=False):
...
+    if was_set_once:
+        return
+
+    if is_screen_kit():
+        rotation = get_screen_value('display_rotate')
+        set_config_value('display_rotate', rotation)
+
+        if not os.path.exists(SYSTEM_FLAGS_DIR):
+            os.makedirs(os.path.dirname(SYSTEM_FLAGS_DIR))
+        open(DISPLAY_ROTATE_GLOBAL_FLAG, 'a').close()
```
Comment (tombettany) [inline bin/kano-settings-onboot]: Maybe wrap in a try in case permissions or something causes this to traceback and not run the rest of the script.

Comment (radujipa) [reply bin/kano-settings-onboot]: Wrapped the whole function 👍

**Pull request merged**